scripts/mk_util.py

```diff
@@ -769,7 +769,7 @@ def parse_options():
 # Return a list containing a file names included using '#include' in
 # the given C/C++ file named fname.
 def extract_c_includes(fname):
```

Comment:
I wonder, this code dates back to a time when we didn't require include paths to be relative to src.

Reply:
Oh, like every dir in the source tree was specified as an include path?

Reply:
It used to be the case that includes were all relative to the paths given by the dependencies in mk_project.
This was changed so that include paths are relative to src directory. 
The python build project was written based on the earlier assumption. It couldn't deal with two header files with the same name. So now I added two occurrences of seq_axioms.h (and seq_eq_solver.h) but under different paths. 
This appears to the source of the warning and based on your changes it would seem that this original python code bakes in the unique name assumption.

Reply:
Yep, this all aligns with what I found when I made the change.

I wonder if a different approach would be just to gut this logic all together and just use the path, as it appears in the `#include`?

Reply:
Yes, better if it is possible to remove this customization from the build system.
Sorry, I haven't dug into this in more detail and tried yet.

Reply:
Just so I understand (and this isn't said in a grumpy way, it is to avoid wasted effort): are you saying you'd rather not merge this at all, and await a PR that fixes it "properly"?

That's totally fine, just want to make sure we're on the same page!